src/server.cpp

```diff
@@ -666,6 +666,12 @@ void Server::AsyncRunStep(bool initial_step)
 			-1);
 	}
 
+	/*
+		Delete from memory blocks removed by deleteBlocks() only when pointers
+		to them are (probably) no longer in use
+	*/
+	m_env->getServerMap().deleteDeletedBlocks();
```

Comment:
serverMap should have a step function where you can implement this

Reply:
Added a step function.